### PR TITLE
Create root schema without the "metadata" schema 

### PR DESCRIPTION
**Pull request by vkorukanti**
Root schema returned by `Frameworks.createRootSchema` contains `metadata` subschema. For some clients `metadata` schema is not needed. For example in Drill metadata is managed by INFORMATION_SCHEMA which is added as subschema to root schema.

Is it a good idea to pass a flag to `Frameworks.createRootSchema` to indicate whether `metadata` schema is expected or not in root schema?

### CODE REVIEW

**Comment by julianhyde** [comment]
Yes. Make the flag mandatory, i.e. don't let's have two createRootSchema methods.

File: core/src/main/java/net/hydromatic/optiq/jdbc/OptiqSchema.java [hunk 1]
```diff
@@ -103,12 +103,15 @@ public SubSchemaCache build() {
         };
   }
 
-  /** Creates a root schema. It contains a "metadata" schema containing
-   * definitions of tables, columns etc. */
-  public static OptiqRootSchema createRootSchema() {
+  /** Creates a root schema. When {@paramref addMetadataSchema} is true a
```
Comment (julianhyde) [inline core/src/main/java/net/hydromatic/optiq/jdbc/OptiqSchema.java]: Non-standard javadoc tag, breaks `mvn site`:

```
[ERROR] /Users/jhyde/open1/optiq/core/src/main/java/net/hydromatic/optiq/jdbc/OptiqSchema.java:106: error: unknown tag: paramref
[ERROR] /** Creates a root schema. When {@paramref addMetadataSchema} is true a
```

File: core/src/test/java/net/hydromatic/optiq/tools/FrameworksTest.java [hunk 2]
```diff
@@ -105,6 +105,12 @@ public RelDataType getRowType(RelDataTypeFactory typeFactory) {
         "EnumerableFilterRel(condition=[>($1, 1)])\n"
         + "  EnumerableTableAccessRel(table=[[myTable]])\n"));
   }
+
+  /** Unit test to test create root schema which has no "metadata" schema. */
+  @Test public void testCreateRootSchemaWithNoMetadataSchema() {
+    SchemaPlus rootSchema = Frameworks.createRootSchema(false);
+    assertTrue(rootSchema.getSubSchemaNames().size() == 0);
```
Comment (julianhyde) [inline core/src/test/java/net/hydromatic/optiq/tools/FrameworksTest.java]: please use modern junit:

``` java
assertThat(rootSchema.getSubSchemaNames().size(), equalTo(0));
```

**Comment by vkorukanti** [comment]
Sorry about the invalid javadoc tag. Pushed a new commit to address review comments. `mvn site` ran without any issues.

**Comment by julianhyde** [comment]
Rebased as https://github.com/julianhyde/optiq/commit/57ea2be640af1c835d7b69ceaa045b90e0db3c92.

**Pull request closed without merging**